### PR TITLE
[TASK] Add a "Quick start" tutorial for contributors (DevDays)

### PR DESCRIPTION
**Pull request by garvinhicking**

### CODE REVIEW

**Comment by garvinhicking** [comment]
**Note: I couldn't get the "EXT:styleguide" Frontend Pagetree to work with this DDEV setup. Don't know if it's a bug in the styleguide setup, or a problem with ddev.**

File: Documentation/Quickstart/1-Preqrequisites.rst [hunk 1]
```diff
@@ -0,0 +1,67 @@
+:navigation-title: Prerequisites
+.. include:: /Includes.rst.txt
+
+.. index::
+   single: Prerequisites
+
+.. _quickstart-prerequisites:
```
Comment (froemken) [inline Documentation/Quickstart/1-Preqrequisites.rst:7]:
```suggestion
..  include:: /Includes.rst.txt

..  index::
    single: Prerequisites

..  _quickstart-prerequisites:
```

File: Documentation/Quickstart/1-Preqrequisites.rst [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    Your operating system needs to provide a Bash terminal, many steps of this guide
+    will be executed on the shell.
+
+5.  Git client
+
+    This guide expects you can execute `git` terminal commands.
+
+    This guide will use the *https* method to connect to GitHub, so that you do not
+    neccessarily need a GitHub account.
```
Comment (froemken) [inline Documentation/Quickstart/1-Preqrequisites.rst:44]:
```suggestion
    This guide uses the *https* method to connect to GitHub, so you do not
    need to have a GitHub account.
```

File: Documentation/Quickstart/1-Preqrequisites.rst [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+6.  SSH client plus SSH key(s) and an email account
+
+    This guide expects you can execute `ssh` terminal commands and connect to
+    foreign hosts. You will need a private SSH key pair and know how to authenticate
+    with it. You need an email account to setup accounts.
+
+7.  PHP IDE / Editor
+
+    It is recommended to use a good PHP IDE. As part of the target audience of this guide
+    you should already use something like PhpStorm, Visual Studeio Code, vi(m)...
```
Comment (froemken) [inline Documentation/Quickstart/1-Preqrequisites.rst:55]:
```suggestion
    you should already use something like PhpStorm, Visual Studio Code, vi(m)...
```

File: Documentation/Quickstart/2-Accounts.rst [hunk 4]
```diff
@@ -0,0 +1,44 @@
+:navigation-title: Accounts
+.. include:: /Includes.rst.txt
+
+.. index::
+   single: Accounts
+
+.. _quickstart-accounts:
```
Comment (froemken) [inline Documentation/Quickstart/2-Accounts.rst:7]:
```suggestion
..  include:: /Includes.rst.txt

..  index::
    single: Accounts

..  _quickstart-accounts:
```

File: Documentation/Quickstart/2-Accounts.rst [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    Vital account! Acts as a Single Sign On (SSO) provider for
+    several other services. See :ref:`TYPO3Account`.
+
+2.  Gerrit (`https://review.typo3.org`__)
+
+    The code review system that is used for your code contributions.
+    There, all patches are commented and voted/approved/rejected.
+
+    You login here via the `My TYPO3` SSO account and need to upload
+    your SSH public key here. See :ref:`GerritAccount`.
```
Comment (froemken) [inline Documentation/Quickstart/2-Accounts.rst:27]: Two `here` in one sentence. What about?

```suggestion
    First login via the `My TYPO3` SSO account. Afterwards you need to upload
    your SSH public key here. See :ref:`GerritAccount`.
```

File: Documentation/Quickstart/3-Git.rst [hunk 6]
```diff
@@ -0,0 +1,48 @@
+:navigation-title: Git
+.. include:: /Includes.rst.txt
+
+.. index::
+   single: Git repository
+
+.. _quickstart-git:
```
Comment (froemken) [inline Documentation/Quickstart/3-Git.rst:7]:
```suggestion
..  include:: /Includes.rst.txt

..  index::
    single: Git repository

..  _quickstart-git:
```

File: Documentation/Quickstart/4-DDEV.rst [hunk 7]
```diff
@@ -0,0 +1,75 @@
+:navigation-title: DDEV
+.. include:: /Includes.rst.txt
+
+.. index::
+   single: DDEV
+
+.. _quickstart-ddev:
```
Comment (froemken) [inline Documentation/Quickstart/4-DDEV.rst:7]:
```suggestion
..  include:: /Includes.rst.txt

..  index::
    single: DDEV

..  _quickstart-ddev:
```

File: Documentation/Quickstart/4-DDEV.rst [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+Quick Start: Set up DDEV
+========================
+
+..  rst-class:: bignums-xxl
+
+1.  Set up DDEV with default convention
+
+    ..  code:: bash
+
+        chdir ~/TYPO3-Contribute
```
Comment (froemken) [inline Documentation/Quickstart/4-DDEV.rst:18]: I have never ever worked with `chdir`. Either on Mac, nor on Linux nor on Windows. Just `cd`.

File: Documentation/Quickstart/4-DDEV.rst [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+        ddev config \
+            --project-name 't3dev' \
+            --project-type 'typo3' \
+            --docroot '.' \
+            --database 'mariadb:10.11' \
+            --php-version '8.3' \
+            --composer-version 'stable' \
+            --nodejs-version '20' \
+            \
+            --project-tld 'ddev.site' \
```
Comment (froemken) [inline Documentation/Quickstart/4-DDEV.rst:28]: `ddev.site` is default

File: Documentation/Quickstart/4-DDEV.rst [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+            --router-http-port '80' \
+            --router-https-port '443' \
+            --additional-hostnames 't3dev-staging.ddev.site,t3dev-production.ddev.site' \
+            --webserver-type 'apache-fpm' \
```
Comment (froemken) [inline Documentation/Quickstart/4-DDEV.rst:32]: Just a question: Why prefer apache?

File: Documentation/Quickstart/5-TYPO3.rst [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+4.  Log in to TYPO3
+
+    You should now be able to log in to the TYPO3 backend with the username and
+    password as specified above (`john-doe : John-Doe-1701D.`) by calling:
+
+    ..  code::
+
+        ddev launch typo3
+
+    .. attention::
```
Comment (froemken) [inline Documentation/Quickstart/5-TYPO3.rst:93]:
```suggestion
    ..  attention::
```

File: Documentation/Quickstart/6-Patch-Intermezzo.rst [hunk 12]
```diff
@@ -0,0 +1,157 @@
+:orphan:
+:navigation-title: Create patch - Intention
+.. include:: /Includes.rst.txt
+
+.. index::
+   single: Patch
+
+.. _quickstart-intent:
```
Comment (froemken) [inline Documentation/Quickstart/6-Patch-Intermezzo.rst:8]:
```suggestion
..  include:: /Includes.rst.txt

..  index::
    single: Patch

..  _quickstart-intent:
```

File: Documentation/Quickstart/6-Patch-Intermezzo.rst [hunk 13]
```diff
@@ -0,0 +1,157 @@
...
+
+This could be many things:
+
+*  **Fix a bug** in one of the TYPO3 Core Extensions
+*  Introduce a **new feature**
+*  Fix mistakes or add content in the **Documentation** of any of the Core Extensions
+*  Enhance **tests** for certain functionality
+*  Raise or add **dependencies** within TYPO3
+*  Adjust the **TYPO3 Build** or **Testing Pipeline**
+*  Make changes to the **TypeScript/SCSS build** of TYPO3
```
Comment (froemken) [inline Documentation/Quickstart/6-Patch-Intermezzo.rst:23]:
```suggestion
*   **Fix a bug** in one of the TYPO3 Core Extensions
*   Introduce a **new feature**
*   Fix mistakes or add content in the **Documentation** of any of the Core Extensions
*   Enhance **tests** for certain functionality
*   Raise or add **dependencies** within TYPO3
*   Adjust the **TYPO3 Build** or **Testing Pipeline**
*   Make changes to the **TypeScript/SCSS build** of TYPO3
```

File: Documentation/Quickstart/6-Patch.rst [hunk 14]
```diff
@@ -0,0 +1,178 @@
+:navigation-title: Create patch
+.. include:: /Includes.rst.txt
+
+.. index::
+   single: Patch
+
+.. _quickstart-patch:
+
```
Comment (froemken) [inline Documentation/Quickstart/6-Patch.rst:8]:
```suggestion
..  include:: /Includes.rst.txt

..  index::
    single: Patch

..  _quickstart-patch:

```

File: Documentation/Quickstart/6-Patch.rst [hunk 15]
```diff
@@ -0,0 +1,178 @@
...
+    title after creation, or in the URL (for example: `12345`).
+
+2.  Ensure proper Git state
+
+    Make sure you are in the right directory and following the previous
+    steps, you should not have any other modified files:
+
+    ..  code:: bash
+
+        cd ~/TYPO3-Contribute
```
Comment (froemken) [inline Documentation/Quickstart/6-Patch.rst:46]: Good to know you also know `cd` :-) Just kidding ;-)

File: Documentation/Quickstart/6-Patch.rst [hunk 16]
```diff
@@ -0,0 +1,178 @@
...
+
+    Depending on the feedback you may need to further refine your patch.
+
+    You can do this by locally editing your files. Then it is **vital**
+    that you do not perform a new git commit, but always only **amend your commit**:
+
+    ..  code:: bash
+
+        // ... edit files
+        git commit . --amend
```
Comment (froemken) [inline Documentation/Quickstart/6-Patch.rst:162]: Is the `.` really needed? Have never used it.

File: Documentation/Quickstart/7-Contribute.rst [hunk 17]
```diff
@@ -0,0 +1,76 @@
+:navigation-title: Reviewing & more
+.. include:: /Includes.rst.txt
+
+.. index::
+   single: Reviewing and more contribution
+
+.. _quickstart-contribute:
```
Comment (froemken) [inline Documentation/Quickstart/7-Contribute.rst:7]:
```suggestion
..  include:: /Includes.rst.txt

..  index::
    single: Reviewing and more contribution

..  _quickstart-contribute:
```

File: Documentation/Quickstart/7-Contribute.rst [hunk 18]
```diff
@@ -0,0 +1,76 @@
...
+not taken up much of your time.
+
+With your environment, you can:
+
+..  rst-class:: bignums-xxl
+
+1.  Keep up to date with development
+
+    By executing this in your working environment,
+    you can always catch up the the latest `main`
```
Comment (froemken) [inline Documentation/Quickstart/7-Contribute.rst:23]:
```suggestion
    you can always catch up the latest `main`
```

File: Documentation/Quickstart/Index.rst [hunk 19]
```diff
@@ -0,0 +1,159 @@
+:navigation-title: Quick Start: 15 Mins to contribute!
+
+.. include:: /Includes.rst.txt
+
+.. index::
+   single: Quickstart, Kickstart
+
+.. _quickstart:
```
Comment (froemken) [inline Documentation/Quickstart/Index.rst:8]:
```suggestion
..  include:: /Includes.rst.txt

..  index::
    single: Quickstart, Kickstart

..  _quickstart:
```

**Review by froemken — CHANGES_REQUESTED** [review]

File: Documentation/Quickstart/6-Patch.rst (hunk 16 above)
Comment (garvinhicking) [reply Documentation/Quickstart/6-Patch.rst:161]: It's an alternate for "-a", I muscle memory this. I'll let @sbuerk decide ;)

**Review by froemken — APPROVED** [review]
Good work John Doe ;-)

File: Documentation/Quickstart/6-Patch.rst (hunk 16 above)
Comment (sbuerk) [reply Documentation/Quickstart/6-Patch.rst:174]: The create patch intermezzo chapter splitts this by adding files with git add first. The `.` looks kind weired here, and I would more switch to `git commit -a --amend` here - or enhance the example here to to use `git add ...` to add specific files and than use only `git commit --amend` here.

Personally I use 

```
git commit --amend -a
```

through an alias - falling back to 

```
git commit --amend
```

to only commit staged files, keeping other files anstaged or make followup changes.

**Comment by sbuerk** [comment]
> **Note: I couldn't get the "EXT:styleguide" Frontend Pagetree to work with this DDEV setup. Don't know if it's a bug in the styleguide setup, or a problem with ddev.**

I changed the todo in the PR and the command to generate the `sys_templtae` based frontend for now until the Site Sets issue is fixed. That makes it not so scary for contributors for now, while still mentioning something if they want to work with Site Sets.

File: Documentation/Quickstart/6-Patch.rst (hunk 16 above)
Comment (garvinhicking) [reply Documentation/Quickstart/6-Patch.rst:174]: I'm fine with switching/changing this, please feel free to adjust so that it works for you!

**Comment by garvinhicking** [comment]
> Really great work @garvinhicking and thank you for jumping in after a short pitch for that. That helps us a lot, I'd say.
> 
> However, pushed some minior things to slightly improve it - at least in my eyes. Hope that is fine for you !

Thank you! I think there are very many improvements to this. I did try to rework it slightly with some follow-up commits.

Let's have a chat to finalize it. Working on it together is always a win in my book, even if there are differences in the details, we can work that out :-)

**Comment by linawolf** [comment]
Wow, this was a good piece of work, thanks a lot for taking care of it!

**Review by sbuerk — APPROVED** [review]
+1 from my side for know, will ping Benni on that pull-request if he wanna have a look before merge.

**Comment by sbuerk** [comment]
Benni does not have time for it right now (we are all quite busy), but we have the approval to go on and merge it for now if we are confident. He will have a look at a later time.

So I'm goint to merge it now as a really good first improvement, and a really big thanks to garvin for the time putted into this improvement :heart: !

**Pull request merged**